### PR TITLE
[Bombastic Perks] Add more perks

### PR DESCRIPTION
**Pull request by Standing-Storm**
#### Summary
Mods "[Bombastic Perks] Add more perks"

#### Purpose of change

Had yet more ideas for perks

#### Describe the solution

Add some perks:

- Grit Your Teeth: Activate to reduce your current pain by 30 or by half (whichever is greater) for 3 minutes. When it ends, all previous pain returns. The cooldown is minutes equal to your pain at the time you used the perk. _Requires Built Tough; cannot have Pain Sensitive or its upgrades_
- Undying Loyalty: If you take damage when your torso or head is at 50% health or lower, and you have any allies with you, all nearby allies take 20% less damage for 1 minute. Can trigger no more than once per 5 minutes.  _Requires Extrovert, cannot have the various predatory mind mutations (Pack Hunter is allowed)_
- Non-combatant: Hey, that's against the Geneva Convention! You take 25% less damage when not wielding a weapon. _Requires Pacifist trait_

And one playstyle perk:
- Moonstruck: You are greatly affected by the phases of the moon. When the moon is full, you gain +3 to all stats. When the moon is waxing, you gain +1 to all stats, when the moon is waning, you gain no benefit, and during the new moon you suffer a -2 penalty to all stats.  (done as a recurring EoC that rechecks the buff every hour)

More to come.

I also changed the enchant on Built Tough to ARMOR_ALL, since it explicitly says it's supposed to reduce all damage by 5%.

#### Testing

Tested perks, they all require appropriate prerequisites and all work properly.

### CODE REVIEW

File: data/mods/BombasticPerks/perkmenu.json [hunk 1]
```diff
@@ -521,6 +521,30 @@
...
+        "text": "Gain [<trait_name:perk_grit_your_teeth>]",
+        "effect": [
+          { "set_string_var": "<trait_name:perk_grit_your_teeth>", "target_var": { "context_val": "trait_name" } },
+          {
+            "set_string_var": "<trait_description:perk_grit_your_teeth>",
+            "target_var": { "context_val": "trait_description" }
+          },
+          { "set_string_var": "perk_grit_your_teeth", "target_var": { "context_val": "trait_id" } },
+          {
+            "set_string_var": "Must have the Built Tough perk./n/nMust not have the Pain Sensitive, Hyperalgesia, or Extreme Hyperalgesia Traits",
```
Comment (Qrox) [inline data/mods/BombasticPerks/perkmenu.json:535]:
```suggestion
            "set_string_var": "Must have the Built Tough perk.\n\nMust not have the Pain Sensitive, Hyperalgesia, or Extreme Hyperalgesia Traits",
```

**Comment by OtpalTS** [comment]
Doesnt this mean that Non-combatant is a -15% damage reduction for unarmed?

**Comment by Standing-Storm** [comment]
> Doesnt this mean that Non-combatant is a -15% damage reduction for unarmed?

Yes. Since without mods unarmed does almost no damage anyway, I don't think it's particularly unbalanced.

**Comment by ashGlaw** [comment]
Seems legit to me

**Comment by EternalEmo** [comment]
Description for non-combatant is still 15% instead of 25%?

**Comment by Standing-Storm** [comment]
…so it is. The actual effect is 25%. I still have more perk ideas—I’ll fix it in the next PR if no one gets to it before then.

**Pull request merged**